scripts/update_repos_version.py

```diff
@@ -0,0 +1,153 @@
+    'https://github.com/edx/xqueue.git',
+    'https://github.com/edx/edx-platform.git',
+)
+
+# Here's where we'll clone the repos.
+TEMP_DIRECTORY = 'git_tmp'
+
+RELEASE_UTIL_REPO_NAME = 'edx-django-release-util'
+
+
+class GitHubApiUtils(object):
+    """
+    Class to query/set GitHub info.
+    """
+    def __init__(self, repo_id):
+        """
+        Returns a GitHub object, possibly authed as a user.
+        """
+        token = os.environ.get('GITHUB_TOKEN', '')
+        username = os.environ.get('GITHUB_USERNAME', '')
+        password = os.environ.get('GITHUB_PASSWORD', '')
+        if len(token):
+            self.gh = github.Github(login_or_token=token)
+        elif len(username) and len(password):
+            self.gh = github.Github(login_or_token=username, password=password)
+        else:
+            # No auth available - use the API anonymously.
+            self.gh = github.Github()
+        self.repo = self.gh.get_repo(repo_id)
+
+    def create_pull(self, *args, **kwargs):
+        return self.repo.create_pull(*args, **kwargs)
+
+
+@click.command()
+@click.option("--new_version", help="New version of edx-django-release-util.", type=str, required=True)
+def bump_repos_version(new_version):
+    """
+    Changes the pinned version number in the requirements files of all repos
+    which have edx-django-release-util as a dependency.
+
+    This script assumes the following environment variables are set for GitHub authentication:
+    Either: GITHUB_TOKEN -or- GITHUB_USERNAME & GITHUB_PASSWORD.
+    """
+    # Make the cloning directory and change directories into it.
+    original_dir = os.getcwd()
+    if not os.path.exists(TEMP_DIRECTORY):
+        os.makedirs(TEMP_DIRECTORY)
+    os.chdir(TEMP_DIRECTORY)
+    tmp_dir = os.getcwd()
+
+    # Iterate through each repository.
+    for repo in REPOS_TO_CHANGE:
+        repo_pattern = re.compile(r'^https://github.com/edx/(.+).git$')
+        repo_name = repo_pattern.match(repo).groups()[0]
+
+        gh = GitHubApiUtils('edx/{}'.format(repo_name))
+
+        # Clone the repo.
+        ret_code = subprocess.call(['git', 'clone', repo])
+        if ret_code:
+            err_msg = "Failed to clone repo {}".format(repo)
+            logging.error(err_msg)
+            raise Exception(err_msg)
+
+        # Change into the cloned repo dir.
+        os.chdir(repo_name)
+
+        # Create a branch, using the version number.
+        branch_name = '{}/{}'.format(RELEASE_UTIL_REPO_NAME, new_version)
+        ret_code = subprocess.call(['git', 'checkout', '-b', branch_name])
+        if ret_code:
+            err_msg = "Failed to create branch in repo {}".format(repo)
+            logging.error(err_msg)
+            raise Exception(err_msg)
+
+        # Search through all TXT files to find all lines with edx-django-release-util, changing the pinned version.
+        for root, dirs, files in os.walk('.'):
+            for file in files:
+                if file.endswith('.txt'):
```

Comment:
Some repos might have huge, unrelated .txt files, and we won't have control over that.
Could we rely on some convention/regex?
Maybe the files will always be under a folder named "requirements" or have some part of the word "requirements" in the filename?